File: Easy/118. Pascal's Triangle/118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []

        for i in range(numRows):
            row = []

            for j in range(i + 1):
                if j == 0 or j == i:
                    row.append(1)
                else:
                    # Calculate the value using the formula: C(n, k) = C(n-1, k-1) + C(n-1, k)
                    value = result[i - 1][j - 1] + result[i - 1][j]
                    row.append(value)

            result.append(row)

        return result

File: Easy/118. Pascal's Triangle/118. Pascal's Triangle.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        if num_rows == 0 {
            return Vec::new();
        }

        let mut result = Vec::new();
        result.push(vec![1]); // The first row is always [1].

        for i in 1..num_rows as usize {
            let prev_row = &result[i - 1];
            let mut new_row = Vec::new();

            new_row.push(1); // The first element is always 1.

            for j in 1..i {
                // Calculate the value using the formula: C(n, k) = C(n-1, k-1) + C(n-1, k)
                let value = prev_row[j - 1] + prev_row[j];
                new_row.push(value);
            }

            new_row.push(1); // The last element is always 1.
            result.push(new_row);
        }

        result
    }
}
